test/Tokenizer.t.sol

```diff
@@ -198,22 +210,56 @@ contract TokenizerTest is Test {
         assertEq(tokenContract.balanceOf(bob), 10_000);
     }
 
-    // function testCanUpgradeErc20TokenImplementation() public {
-    //     vm.startPrank(deployer);
-    //     vm.stopPrank();
+    function testCanUpgradeErc20TokenImplementation() public {
+        vm.selectFork(mainnetFork);
 
-    //     vm.startPrank(originalOwner);
-    //     IPToken tokenContractOld = tokenizer.tokenizeIpnft(1, 100_000, "IPT", agreementCid, "");
-    //     vm.stopPrank();
+        address mainnetOwner = 0xCfA0F84660fB33bFd07C369E5491Ab02C449f71B;
+        address mainnetTokenizer = 0x58EB89C69CB389DBef0c130C6296ee271b82f436;
+        address mainnetIPNFT = 0xcaD88677CA87a7815728C72D74B4ff4982d54Fc1;
 
-    //     vm.startPrank(deployer);
-    //     TokenizerNext synthNext = new TokenizerNext();
-    //     tokenizer.upgradeTo(address(synthNext));
-    //     TokenizerNext synth2 = TokenizerNext(address(tokenizer));
+        Tokenizer upgradedTokenizer = Tokenizer(mainnetTokenizer);
+        IPNFT ipnftMainnetInstance = IPNFT(mainnetIPNFT);
+
+        IPToken newIPTokenImplementation = new IPToken();
+        
+        vm.startPrank(mainnetOwner); // Owner address on mainnet
+        tokenizer11 = Tokenizer11(mainnetTokenizer);
+        tokenizer11.upgradeTo(address(new Tokenizer()));
+
+        upgradedTokenizer.setIPTokenImplementation(address(newIPTokenImplementation));
+        assertEq(upgradedTokenizer.ipTokenImplementation(), address(newIPTokenImplementation));
+        
+        IPermissioner _permissioner = new BlindPermissioner();
+        upgradedTokenizer.reinit(_permissioner); // project is already initialized
+        vm.stopPrank();
+
+        address ipnftHolder = 0xD920E60b798A2F5a8332799d8a23075c9E77d5F8;
+        console.log(ipnftMainnetInstance.ownerOf(3));
+        assertEq(ipnftMainnetInstance.ownerOf(3), ipnftHolder);
+        
+        vm.startPrank(ipnftHolder);
+        IPToken createdIPToken =  upgradedTokenizer.tokenizeIpnft(3, 100_000, "IPT", agreementCid, "");
+        assertEq(createdIPToken.balanceOf(ipnftHolder), 100_000);
+        createdIPToken.transfer(alice, 1);
+        assertEq(createdIPToken.balanceOf(ipnftHolder), 99_999);
+        vm.stopPrank();
+
+        address vitaFastHolder = 0x45602BFBA960277bF917C1b2007D1f03d7bd29e4;
+        IPToken vitaFast = IPToken(0x6034e0d6999741f07cb6Fb1162cBAA46a1D33d36);
+       
+        vm.startPrank(vitaFastHolder);
+        assertEq(vitaFast.balanceOf(vitaFastHolder), 16942857059768483219100);
+        vitaFast.transfer(alice, 100);
+        assertEq(vitaFast.balanceOf(vitaFastHolder), 16942857059768483219000);
+        vm.stopPrank();
+        
+        //IPToken exampleVitaFast = IPToken(0x6034e0d6999741f07cb6Fb1162cBAA46a1D33d36);
+        //console.log(exampleVitaFast.balanceOf(vitaUser));
+        //exampleVitaFast.transfer(alice, 100 wei);
+
+
+        //vm.stopPrank();
```

Comment:
add a test that the IPNFT#2 holder (a multisig) can call `issue` and `cap` on the current Vitafast IPToken instance and that the metadata is still ok (ipnftId = 2, originalOwner = ipnft#2 holder). 

Test that this also works / holds true for the new IPToken that has been spawned by the IPNFT#3 holder